web/src/reducers/archive-retro-middleware.js

```diff
@@ -0,0 +1,60 @@
+/*
+ * Postfacto, a free, open-source and self-hosted retro tool aimed at helping
+ * remote teams.
+ *
+ * Copyright (C) 2016 - Present Pivotal Software, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ *
+ * it under the terms of the GNU Affero General Public License as
+ *
+ * published by the Free Software Foundation, either version 3 of the
+ *
+ * License, or (at your option) any later version.
+ *
+ *
+ *
+ * This program is distributed in the hope that it will be useful,
+ *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *
+ * GNU Affero General Public License for more details.
+ *
+ *
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ *
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+const ArchiveMiddleware = (actionDispatcher) => (store) => (next) => (action) => {
```

Comment:
I found this a little tricky to follow, maybe something like:
```js
const ArchiveMiddleware = (actionDispatcher) => (store) => (next) => (action) => {
  if (action.type === 'CURRENT_RETRO_ITEM_DONE_UPDATED') {
    const {itemId, done} = action.payload;
    const currentRetro = store.getState().retro.currentRetro;

    if (done) {
      const item = currentRetro.items.find((i) => i.id === itemId);
      actionDispatcher.completedRetroItemAnalytics({data: {retroId: currentRetro.id, category: item.category}});
    }

    const allDone = currentRetro.items
      .filter((item) => item.id !== itemId)
      .every((i) => i.done);

    if (allDone) {
      store.dispatch({
        type: 'SHOW_DIALOG',
        payload: {
          title: 'Archive this retro?',
          message: 'The board will be cleared ready for your next retro and incomplete action items will be carried across.',
        },
      });
    }
  }
  next(action);
};
```

Reply:
I think that changes the check because the item you filter out may not be done now. This means not all retro items would be done. Do agree it's confusing though but not seeing a better alternative right now.

Reply:
Ah, I did wonder if there was a case I wasn't spotting. I don't think it's covered by the tests..

Reply:
Ah, you're right, I'll back fill one in the next PR